### PR TITLE
refactor(cli): print relevant values when comparing fails

### PR DESCRIPTION
**Pull request by baumstern**
## Better Logging

Before: only error message has been printed
After: error message and relevant values will be printed

### CODE REVIEW

**Comment by gurrpi** [comment]
> @gurrpi what's the status on this? Do you think this is worth finishing/merging? Or should we close this out?

I've just triggered a new CI run. Once the checks pass, I think this is worth merging.

**Comment by samajammin** [comment]
Nudge on this @baumstern

**Comment by baumstern** [comment]
@ctrlc03 PTAL

File: cli/ts/utils.ts [hunk 1]
```diff
@@ -142,6 +142,21 @@ const isPathExist = (paths: Array<string>): [boolean, string] => {
     return [true, null]
 }
 
+const CompareOnchainValue = (
```
Comment (samajammin) [inline cli/ts/utils.ts:145]:
```suggestion
const compareOnchainValue = (
```
General convention - functions should always be camelCase, ya?

Noting this commit suggestion would require updating all uses of this function...

File: cli/ts/utils.ts [hunk 2]
```diff
@@ -142,6 +142,21 @@ const isPathExist = (paths: Array<string>): [boolean, string] => {
...
+    name: string,
+    onChainValue: any,
```
Comment (samajammin) [inline cli/ts/utils.ts:147]: Another naming convention nitpick - here you capitalize the "C" in `onChain` but you didn't in the function name. I suggest we be consistent (i.e. use capital "C" for both)

File: cli/ts/utils.ts (hunk 1 above)
Comment (samajammin) [reply cli/ts/utils.ts:145]: Edit - also capitalized the "C". See comment below for context

**Comment by samajammin** [comment]
@baumstern any particular reason you PR'd from your fork vs. creating a branch directly off the repo?

File: cli/ts/proveOnChain.ts [hunk 3]
```diff
@@ -380,6 +375,7 @@ const proveOnChain = async (args: any) => {
             return 
         }
 
+        const txErr = 'Error: processMessages() failed'
```
Comment (samajammin) [inline cli/ts/proveOnChain.ts:378]: Another consistency nitpick - looks like you're using a mix of single quotes & double quotes. Do we have a convention on this? I don't have a strong preference... just that we be consistent

File: cli/ts/utils.ts [hunk 4]
```diff
@@ -142,6 +142,21 @@ const isPathExist = (paths: Array<string>): [boolean, string] => {
...
+    offChainValue: any,
+): boolean => {
+    if (onChainValue !== offChainValue) {
+        console.error(`Error: ${name} mismatch.`)
+        console.error("     onChainValue: " + onChainValue)
```
Comment (samajammin) [inline cli/ts/utils.ts:152]: Why the spacing on these logs?

**Review by samajammin — CHANGES_REQUESTED** [review]
A few styling nitpicks - see my comments

File: cli/ts/utils.ts (hunk 1 above)
Comment (baumstern) [reply cli/ts/utils.ts]: > General convention - functions should always be camelCase, ya?
> 
> Noting this commit suggestion would require updating all uses of this function...

Good catch! Reflected in [974ab46](https://github.com/privacy-scaling-explorations/maci/pull/585/commits/974ab462a55ad39f0df76ce16972c53446480589)

File: cli/ts/utils.ts (hunk 2 above)
Comment (baumstern) [reply cli/ts/utils.ts]: Having seen `onchain` used more commonly, I've decided to go with that. Thanks for the input!

File: cli/ts/utils.ts (hunk 4 above)
Comment (baumstern) [reply cli/ts/utils.ts]: The spacing in these logs is intended to make the output more noticeable and easier to read.

**Comment by baumstern** [comment]
Inconsistency has inherited from legacy. Filed an issues for that
* https://github.com/privacy-scaling-explorations/maci/issues/811
* https://github.com/privacy-scaling-explorations/maci/issues/810

> @baumstern any particular reason you PR'd from your fork vs. creating a branch directly off the repo?

I think I didn't dare to add a branch to the upstream repo at that moment 😂

**Comment by baumstern** [comment]
> A few styling nitpicks - see my comments

The inconsistency has been inherited from legacy code. I believe it would be better to address the codebase's overall consistency, so I have filed separate issues to tackle this:

    [Issue 811](https://github.com/privacy-scaling-explorations/maci/issues/811)
    [Issue 810](https://github.com/privacy-scaling-explorations/maci/issues/810)

**Comment by samajammin** [comment]
> The inconsistency has been inherited from legacy code

I'm just referring to your changes in this PR - you use both `Onchain` & `OnChain`, as well as single/double quotes. So I'd start by just addressing this PR. i.e. be consistent with your individual commits.

But agree, would be good to unify across the codebase as well 👍

**Comment by baumstern** [comment]
> > The inconsistency has been inherited from legacy code
> 
> I'm just referring to your changes in this PR - you use both `Onchain` & `OnChain`, as well as single/double quotes. So I'd start by just addressing this PR. i.e. be consistent with your individual commits.
> 
> But agree, would be good to unify across the codebase as well 👍

That's reasonable. Fixed the inconsitency of this PR in 2e26126 to uses double quotes and `OnChain` across the change.

Lint errors such as those in https://github.com/privacy-scaling-explorations/maci/issues/457 should be fixed and automatically enforced linting in pre-commit hook would ensure us to have consistent style after that.

**Review by samajammin — APPROVED** [review]

**Pull request merged**